src/SoftDeleteable/SoftDeleteableListener.php

```diff
@@ -129,10 +137,27 @@ public function onFlush(EventArgs $args)
                         $postSoftDeleteEventArgs
                     );
                 }
+
+                $this->softDeletedObjects[] = $object;
```

Comment:
More than a suggestion, this is a question or a request for analysis.

Given storing the list of deleted objects in this array could lead to high memory consumption, do you evaluated a way to mitigate this by using the object references (like `spl_object_id(), `spl_object_hash()`, or a combination of the model class and its id)?

I don' t know if this can produce better results, but I think this is a consideration we should take into account.

Reply:
Being that these objects are unset (cleared from memory) `postFlush`, I really don't see it as much of a concern - certainly not enough to introduce `spl_object_id` complexity and the potential to expose bugs through that.

Reply:
These models shouldn't be held in memory for too long anyway short of something catastrophic happening between the `onFlush` event and the `postFlush` event.  So it really boils down to what happens if you try re-using a closed object manager (since the ORM's UoW is closing the EM and rolling back the active transaction on fail at least).